src/services/healthcareProfessionalService.ts

```diff
@@ -324,7 +334,50 @@ function validateNames(
         })
     }
 
-    //TODO validate each name
+    if (names && names[0].firstName.length > 30) {
+        validationResults.hasErrors = true
+        validationResults.errors?.push({
+            field: 'names',
+            errorCode: ErrorCode.INVALID_LENGTH_TOO_LONG,
+            httpStatus: 400
+        })
+    }
+
+    if (names && !names[0].firstName.trim()) {
+        validationResults.hasErrors = true
+        validationResults.errors?.push({
+            field: 'names',
+            errorCode: ErrorCode.REQUIRED,
+            httpStatus: 400
+        })
+    }
+
+    if (names && names[0].lastName.length > 30) {
+        validationResults.hasErrors = true
+        validationResults.errors?.push({
+            field: 'names',
+            errorCode: ErrorCode.INVALID_LENGTH_TOO_LONG,
+            httpStatus: 400
+        })
+    }
+
+    if (names && !names[0].lastName.trim()) {
+        validationResults.hasErrors = true
+        validationResults.errors?.push({
+            field: 'names',
+            errorCode: ErrorCode.REQUIRED,
+            httpStatus: 400
+        })
+    }
+
+    if (names && names[0].middleName && names[0].middleName.length > 60) {
```

Comment:
I have added above 60 since some people have two middle names, for example my mother has two (quite long ones). Not sure if it is needed.